### ISSUE TITLE
[BUG] Configuration inconsitancy! Move common blockchain parameters to genesis.block

### ISSUE THREAD

**Issue by kuvaldini** [hyperledger/iroha#1079]
Let's consider `config.sample`
```
{
  "block_store_path" : "/tmp/block_store/",
  "torii_port" : 50051,
  "internal_port" : 10001,
  "pg_opt" : "host=localhost port=5432 user=postgres password=mysecretpassword",
  "max_proposal_size" : 10,
  "proposal_delay" : 5000,
  "vote_delay" : 5000,
  "mst_enable" : false,
  "mst_expiration_time" : 1440,
  "max_rounds_delay": 3000,
  "stale_stream_max_rounds": 2,
  "metrics": "127.0.0.1:8080"
}
```
Some configuration parameters `proposal_delay` `max_proposal_size` `mst_expiration_time` `mst_enable` and others 
**MUST be the same** in all over the Iroha network. Inconsistency (unequal) between parameters may (should, will) lead Iroha network (some nodes) to undefined behavior and/or to node(s) freeze.

## One of possible solutions 
* make these parameters a part of blockchain and store them in genesis block. 
* provide additional commands to edit configuration parameters

## The other solution 
always synchronize parameters that must always be equal, mark neighbor node INVALID if it has inconsistent parameters and do not negotiate blocks with "invalid" node.

## Suggest your solution pls
if you have ideas

---------
Kind ping @lebdron @kamilsa @iceseer @LiraLemur 

-------
@baziorek, could you please test and give logs from inconsistent Iroha network, i.e. where `max_proposal_size` is different

**Comment by baziorek** [comment]
> @baziorek, could you please test and give logs from inconsistent Iroha network, i.e. where max_proposal_size is different
I'll try to do this on two-noded test network tomorrow in the morning.

**Comment by kamilsa** [comment]
yes, they should be part of genesis block. I would create new commands and permissions that update configuration parameters

**Comment by baziorek** [comment]
> @baziorek, could you please test and give logs from inconsistent Iroha network, i.e. where max_proposal_size is different
Here are results https://gist.github.com/baziorek/6c8d82898f48803b90f27453058eabc5 for 2 nodes with `"max_proposal_size" : 20,` vs `"max_proposal_size" : 10,` it is version iroha 1.2.0. Scenarios:
1. Running 2 nodes with different `"max_proposal_size"` then shutting down - everything looks OK
2. Running 2 nodes with different `"max_proposal_size"` then sending 2 transactions and shutting down - everything looks OK
3. I'll now try to send 21 fast transactions (without waiting for response) then update the post and the gist

**Comment by baziorek** [comment]
@kuvaldini I've created tests. It looks like it works OK when I send transactions to node which accepts less `max_proposal_size`, I'll try to do the opposite and update logs.

**Comment by baziorek** [comment]
@kuvaldini I've performed tests again on version iroha 1.2.1: two-noded network:
https://gist.github.com/baziorek/7bc47e05dd82f83783f7bceee300d8b6
it looks like it works and it allows sometimes it is `"max_proposal_size"` from one node, but next time `"max_proposal_size"` from another node.
In the attached gist are:

1. config1.sample
2. config2.sample
3. genesis.block
4. logs from node 1
5. logs from node 2
6. source code in python
7. output of the code

**Comment by kuvaldini** [comment]
@baziorek thank you for response.
I definitely know that some configurations leaded iroha network to halt. 

Let me check them out from history.

**Comment by baziorek** [comment]
> @baziorek thank you for response. Good Job!
> I definitely know that some configurations leaded iroha network to halt.
> 
> Let me check them out from history.

I've also checked another parameter on three nodes:
1. Node1: `"proposal_delay" : 1000,` and `"max_proposal_size" : 10,`
2. Node2: `"proposal_delay" : 3000,` and `"max_proposal_size" : 20,`
3. Node3: `"proposal_delay" : 5000,` and `"max_proposal_size" : 30,`
and it is also working, this time 8 blocks:
0000000000000002(10txs)  0000000000000003(20txs)  0000000000000004(10txs)  0000000000000005(30txs)  0000000000000006(10txs)  0000000000000007(10txs)  0000000000000008(10txs)  0000000000000009(1tx)

**Comment by appetrosyan** [comment]
Iroha v1 is to be deprecated soon
